lib/Raven/Client.php

```diff
@@ -112,13 +113,26 @@ class Client
     protected $config;
 
     /**
+     * @var HttpAsyncClient The HTTP client
+     */
+    private $httpClient;
+
+    /**
+     * @var MessageFactory The message factory
+     */
+    private $messageFactory;
+
+    /**
      * Constructor.
      *
-     * @param Configuration $config The client configuration
+     * @param Configuration   $config     The client configuration
+     * @param HttpAsyncClient $httpClient The HTTP client
      */
-    public function __construct(Configuration $config)
+    public function __construct(Configuration $config, HttpAsyncClient $httpClient, MessageFactory $messageFactory)
```

Comment:
Sorry but I completely disagree. Typehints are alot more useful ([and I already wrote a lengthy blog post about it](https://engineering.facile.it/blog/eng/visual-debt-typehints/)) and using array as a single point of entry for options doesn't protect us from BC breaks. 

The builder class does that a lot more.